compiler/optimizer/OMRTransformUtil.hpp

```diff
@@ -226,7 +226,8 @@ class OMR_EXTENSIBLE TransformUtil
    static TR::Node *generateArrayElementAddressTrees(
       TR::Compilation *comp,
       TR::Node *arrayNode,
-      TR::Node *offsetNode = NULL);
+      TR::Node *offsetNode = NULL,
+      TR::Node *originatingByteCodeNode = NULL);
```

Comment:
It looks like there are existing calls to `generateArrayElementAddressTrees` that will get the default `originatingByteCodeNode` value of NULL, and end up generating `TR::Node`s without the correct bytecode info.

For example, there is a call in [`OMR::TransformUtil::generateFirstArrayElementAddressTrees`](https://github.com/VermaSh/omr/blob/c875d1733e4345cb44744bdbf5a38ab7920a798e/compiler/optimizer/OMRTransformUtil.cpp#L494-L499) which is itself called from [`OMR::ValuePropagation::transformArrayCloneCall`](https://github.com/VermaSh/omr/blob/c875d1733e4345cb44744bdbf5a38ab7920a798e/compiler/optimizer/ValuePropagationCommon.cpp#L4175-L4176).  That transformation is applied for calls to `Object.clone` if the object is known to be an array.

Prior to this change, the IL trees that resulted from that transformation looked like this:

```
n69n        arraycopy  <arraycopy>[#215  helper Method] [flags 0x400 0x0 ] (Unsigned cannotOverflow forwardArrayCopy noArrayStoreCheckArrayCopy wordElementArrayCopy )  [0x7fa62c37f550] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=3 flg=0xd020
n66n          aladd                                                                           [0x7fa62c37f460] bci=[-1,17,5] rc=1 vc=0 vn=- li=- udi=- nc=2
n23n            ==>aload
n65n            lconst 8 (highWordZero X!=0 X>=0 )                                            [0x7fa62c37f410] bci=[-1,0,3] rc=1 vc=0 vn=- li=- udi=- nc=0 flg=0x4104
n68n          aladd                                                                           [0x7fa62c37f500] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=2
n25n            ==>newarray
n67n            lconst 8 (highWordZero X!=0 X>=0 )                                            [0x7fa62c37f4b0] bci=[-1,0,3] rc=1 vc=0 vn=- li=- udi=- nc=0 flg=0x4104
n64n          lmul                                                                            [0x7fa62c37f3c0] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=2
n62n            i2l                                                                           [0x7fa62c37f320] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=1
n55n              ==>arraylength
n63n            lconst 4 (highWordZero X!=0 X>=0 )                                            [0x7fa62c37f370] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=0 flg=0x4104
```

after this change, they look like this:

```
n71n      treetop                                                                             [0x7f52fa57f5f0] bci=[-1,18,5] rc=0 vc=0 vn=- li=- udi=- nc=1
n69n        arraycopy  <arraycopy>[#215  helper Method] [flags 0x400 0x0 ] (Unsigned cannotOverflow forwardArrayCopy noArrayStoreCheckArrayCopy wordElementArrayCopy )  [0x7f52fa57f550] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=3 flg=0xd020
n66n          aladd                                                                           [0x7f52fa57f460] bci=[-1,0,3] rc=1 vc=0 vn=- li=- udi=- nc=2
n23n            ==>aload
n65n            lconst 8 (highWordZero X!=0 X>=0 )                                            [0x7f52fa57f410] bci=[-1,0,3] rc=1 vc=0 vn=- li=- udi=- nc=0 flg=0x4104
n68n          aladd                                                                           [0x7f52fa57f500] bci=[-1,0,3] rc=1 vc=0 vn=- li=- udi=- nc=2
n25n            ==>newarray
n67n            lconst 8 (highWordZero X!=0 X>=0 )                                            [0x7f52fa57f4b0] bci=[-1,0,3] rc=1 vc=0 vn=- li=- udi=- nc=0 flg=0x4104
n64n          lmul                                                                            [0x7f52fa57f3c0] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=2
n62n            i2l                                                                           [0x7f52fa57f320] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=1
n55n              ==>arraylength
n63n            lconst 4 (highWordZero X!=0 X>=0 )                                            [0x7f52fa57f370] bci=[-1,18,5] rc=1 vc=0 vn=- li=- udi=- nc=0 flg=0x4104
```

In particular, notice the bytecode info. on the `aladd` nodes.